File: 1264-maximum-number-of-words-you-can-type/1264-maximum-number-of-words-you-can-type.c
int canBeTypedWords (char* text, char* brokenLetters) {
    bool broken [26] = {0};
    for (size_t i = 0; brokenLetters [i]; ++i) broken [brokenLetters [i] - 'a'] = true;
    int count = 0;
    const char *p = text;
    bool wordHasBroken = false;
    while (*p) {
        if (*p == ' ') {
            if (!wordHasBroken) ++count;
            wordHasBroken = false;
        } else {
            if (!wordHasBroken && broken [*p - 'a']) wordHasBroken = true;
        }
        ++p;
    }
    if (!wordHasBroken) ++count;
    return count;
}

File: 1264-maximum-number-of-words-you-can-type/1264-maximum-number-of-words-you-can-type.py
class Solution:
    def canBeTypedWords (self, text : str, brokenLetters : str) -> int:
        broken = set (brokenLetters)
        count = 0
        for w in text.split ():
            if not any (ch in broken for ch in w):
                count += 1
        return count